### DISCUSSION TITLE
Sonoff MINIR2 switch enablement - Tasmota (GPIO)

### DISCUSSION THREAD

**Discussion by DCG81** [arendst/Tasmota#12993] [Q&A] [answered]
Hello,

I am new to Sonoff Tasmota world, I have recently flashed Tasmoto to Sonoff MINIR2 and now looking to wire it to my existing switch box. (I am from India)

I had couple of electricians come in and they struggled to get the wiring done as per my requirement which is both the Tasmota Smart switch and the existing physical switch to work, the electricians managed get only the Tasmota smart switch to work and the physical switch never worked.

A friend told me that the physical switch is not working because u need some GPIO settings to be changed in Tasmota.

so for the physical switch to work what settings do I need to change under Tasmota settings ?

**Comment by barbudor** [comment, marked as answer]
Did you flashed and configured the Sonoff MiniR2 according to https://templates.blakadder.com/sonoff_MINIR2.html ?
The provided templates include the proper GPIO configuration to use the external switch input.
Connection diagram with Tasmota would be exactly the same as per Sonoff documentation. Do not depend on the software.

Is your physical switch a rocker switch or a push-button ?

**Comment by DCG81** [comment]
I have just flashed sonoff latest bin file but have not configured the template ? so I should configure this template in the console correct ?

**Reply by DCG81** [reply]
In the console I can see the switch mode is set to 0, so now in the GPIO settings i should select GPIO4 as the switch correct ?
I am using a Indian modular switch

**Reply by barbudor** [reply]
Apply the template, activate it and all pins will be assigned as needed

An "Indian modular switch" doesn't say if it is a rocker switch or a push button (which is a country independant terminology)
- Rocker switch => switchmode 0
- Push Button => switchmode 4

**Reply by DCG81** [reply]
yes is its a Rocker switch.
I have applied the template in the console and below is the gpio setting I notice, do I need to change anything in this

also I am not sure why it says Sonoff basic under module type

![0AC68AAE-CFC4-41E4-A5DF-2F8FD766B31C](https://user-images.githubusercontent.com/87446431/131130779-fdce1d0d-634a-40f5-b9a5-01748df5809f.jpeg)

**Reply by barbudor** [reply]
It looks like you haven't activated the template
![image](https://user-images.githubusercontent.com/2996042/131131298-5594cf1c-6cae-41ca-83c8-4cc00b35202a.png)

**Reply by sfromis** [reply]
You could also select the template in "Configure Module", where it will have module number 0.

**Reply by DCG81** [reply]
thank you both, I have now configured and activated the template...

**Reply by barbudor** [reply]
And the rocker-switch started to rock 🕺

**Reply by DCG81** [reply]
haha I havent checked the rocker switch as of now, since I would need to do the wiring again, will surely update back shortly

**Reply by DCG81** [reply]
Happy to report the rocker switch is working now 👍 
thank you @barbudor for ur time
